apps/sales/views.py

```diff
@@ -154,4 +157,168 @@ def create(self, request, *args, **kwargs):
+
+
+#---------- ONE API - MULTIPLE API CALLS (CRUD OPERATIONS) ---------------
+
+class SaleOrderOneView(generics.GenericAPIView,mi.ListModelMixin,mi.CreateModelMixin,mi.RetrieveModelMixin,mi.UpdateModelMixin,mi.DestroyModelMixin):
+    queryset = SaleOrder.objects.all()
+    serializer_class = SaleOrderSerializer
+
+    def get(self, request, *args, **kwargs):
+            if 'pk' in kwargs:
+                return self.retrieve(request, *args, **kwargs)  # Retrieve a single instance
+            return list_all_objects(self, request, *args, **kwargs)
+    
+    # Handling POST requests for creating
+    def post(self, request, *args, **kwargs):   #To avoid the error this method should be written [error : "detail": "Method \"POST\" not allowed."]
+        return self.create(request, *args, **kwargs)
+
+    def create(self, request, *args, **kwargs):
+        given_data = request.data
+
+        sale_order_data = given_data.pop('sale_order')
+        sale_order_items_data = given_data.pop('sale_order_items')
+        order_attachments_data = given_data.pop('order_attachments')
+        order_shipments_data = given_data.pop('order_shipments')
+        
+        # create data in 'saleorder' model
+        serializer = self.get_serializer(data=sale_order_data)
+        if serializer.is_valid(raise_exception=True):
+            # self.perform_create(serializer)
+            serializer.save()
+            data_1 = serializer.data
+    
+            # create data in 'saleorder_items' model
+            sale_order_id = serializer.data.get('sale_order_id', None) # from Saleorder instance id is fetched
+            add_key_value_to_all_ordereddicts(sale_order_items_data,'sale_order_id',sale_order_id) #in sale_order_id replace sale_order_id from new instance
+            data_2 = create_multi_instance(sale_order_items_data,SaleOrderItemsSerializer)
+
+            # create data in 'order_attachments' model
+            add_key_value_to_all_ordereddicts(order_attachments_data,'order_id',sale_order_id) #in order_id replace sale_order_id from new instance
+            data_3 = create_multi_instance(order_attachments_data,OrderAttachmentsSerializer)
+
+            # create data in 'order_shipments' model
+            provided_val = order_shipments_data.get('order_type_id')
+            get_id = OrderTypes.objects.all()
+            name_id_dictionary = {}
+            for obj in get_id:
+                name = obj.name
+                id = obj.order_type_id
+                name_id_dictionary[name] = id
+            order_shipments_data['order_id'] = sale_order_id
+            order_shipments_data['order_type_id'] = name_id_dictionary.get(provided_val)
+
+            serializer = OrderShipmentsSerializer(data=order_shipments_data)
+            if serializer.is_valid(raise_exception=True):
+                serializer.save()
+                data_4 = serializer.data   
+
+            if data_1 and data_2 and data_3 and data_4:  
```

Comment:
Change the data_1 and data_2 and data_3 and data_4 variables to some meaningful and understandable names.

Reply:
Updated Thulasi